keymanager/src/churp/storage.rs

```diff
@@ -72,6 +85,82 @@ impl Storage {
         Ok(())
     }
 
+    /// Loads and decrypts a secret share, consisting of a polynomial
+    /// and its associated verification matrix.
+    pub fn load_secret_share<G>(
+        &self,
+        churp_id: u8,
+        epoch: EpochTime,
+    ) -> Result<Option<SecretShare<G>>>
+    where
+        G: Group + GroupEncoding,
+    {
+        let key = Self::create_secret_share_key(churp_id);
+        let mut ciphertext = self.storage.get(key)?;
+        if ciphertext.is_empty() {
+            return Ok(None);
+        }
+
+        let share = Self::decrypt_secret_share(&mut ciphertext, churp_id, epoch)?;
+        Ok(Some(share))
+    }
+
+    /// Encrypts and stores the provided secret share, consisting of
+    /// a polynomial and its associated verification matrix.
+    pub fn store_secret_share<G>(
+        &self,
+        share: &SecretShare<G>,
+        churp_id: u8,
+        epoch: EpochTime,
+    ) -> Result<()>
+    where
+        G: Group + GroupEncoding,
+    {
+        let key = Self::create_secret_share_key(churp_id);
+        let ciphertext = Self::encrypt_secret_share(share, churp_id, epoch);
+        self.storage.insert(key, ciphertext)?;
+
+        Ok(())
+    }
+
+    /// Loads and decrypts the next secret share, consisting of a polynomial
+    /// and its associated verification matrix.
+    pub fn load_next_secret_share<G>(
+        &self,
+        churp_id: u8,
+        epoch: EpochTime,
+    ) -> Result<Option<SecretShare<G>>>
+    where
+        G: Group + GroupEncoding,
+    {
+        let key = Self::create_next_secret_share_key(churp_id);
+        let mut ciphertext = self.storage.get(key)?;
+        if ciphertext.is_empty() {
+            return Ok(None);
+        }
+
+        let share = Self::decrypt_secret_share(&mut ciphertext, churp_id, epoch)?;
+        Ok(Some(share))
+    }
+
+    /// Encrypts and stores the provided next secret share, consisting of
+    /// a polynomial and its associated verification matrix.
+    pub fn store_next_secret_share<G>(
+        &self,
+        share: &SecretShare<G>,
+        churp_id: u8,
+        epoch: EpochTime,
+    ) -> Result<()>
+    where
+        G: Group + GroupEncoding,
+    {
+        let key = Self::create_next_secret_share_key(churp_id);
+        let ciphertext = Self::encrypt_secret_share(share, churp_id, epoch);
```

Comment:
They intentionally use the same context, as I had planned to overwrite the old one with the next one. However, currently, I'm not doing that, but instead encrypting and storing it again. But this could be changed.

Since there can be only one handoff per epoch, epochs are unique, and the authentication data (AD) will ensure that the decryption of any incorrect share (share that belongs to an invalid handoff/epoch) will fail.